.github/workflows/quality-assurance.yml

```diff
@@ -53,6 +53,24 @@ jobs:
       - name: Run PHPStan
         run: vendor/bin/phpstan --error-format=github
 
+  rector:
+    name: Run Rector
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 8.3
+          coverage: none
+
+      - name: Install composer dependencies
+        uses: ramsey/composer-install@v3
```

Comment:
Do you mind taking a look at caching on this PR? I think we should be able to cache composer dependencies across all steps here.

Reply:
from ramsey/composer-install
> ramsey/composer-install is a GitHub Action to streamline installation of Composer dependencies in workflows. It installs your Composer dependencies and caches them for improved build times.

So I guess we don't need to manually add any caching.

If we do need to do that, then the following can easily be added. Only need to set a good cache key
```yml
- name: Cache dependencies
  id: composer-cache
  uses: actions/cache@v4
  with:
    path: ./vendor
    key: composer-${{ hashFiles('**/composer.lock') }}
```

Reply:
This is specific to the step though, right? I _think_ we can cache across multiple steps.